rust/lance-core/src/io/writer/statistics.rs

```diff
@@ -746,46 +1841,211 @@ mod tests {
-                        Arc::new(ArrowField::new("max_value", DataType::Int32, false)),
+                        Arc::new(ArrowField::new("max_value", DataType::Int32, true)),
                         Arc::new(Int32Array::from(vec![3, std::i32::MAX])) as ArrayRef,
                     ),
                 ])),
-                // Arc::new(StructArray::from(vec![
-                //     (
-                //         Arc::new(ArrowField::new("null_count", DataType::Int64, false)),
-                //         Arc::new(Int64Array::from(vec![6, 0])) as ArrayRef,
-                //     ),
-                //     (
-                //         Arc::new(ArrowField::new("min_value", DataType::Utf8, true)),
-                //         Arc::new(StringArray::from(vec![Some("aaa"), None])) as ArrayRef,
-                //     ),
-                //     (
-                //         Arc::new(ArrowField::new("max_value", DataType::Utf8, true)),
-                //         Arc::new(StringArray::from(vec![Some("bbb"), None])) as ArrayRef,
-                //     ),
-                // ])),
                 Arc::new(StructArray::from(vec![
                     (
                         Arc::new(ArrowField::new("null_count", DataType::Int64, false)),
-                        Arc::new(Int64Array::from(vec![2, 0])) as ArrayRef,
+                        Arc::new(Int64Array::from(vec![6, 0])) as ArrayRef,
+                    ),
+                    (
+                        Arc::new(ArrowField::new("min_value", DataType::Utf8, true)),
+                        Arc::new(StringArray::from(vec![Some("aaa"), None])) as ArrayRef,
+                    ),
+                    (
+                        Arc::new(ArrowField::new("max_value", DataType::Utf8, true)),
+                        Arc::new(StringArray::from(vec![Some("bbb"), None])) as ArrayRef,
+                    ),
+                ])),
+            ],
+        )
+        .unwrap();
+
+        assert_eq!(batch, expected_batch);
+
+        // Check boolean, binary and string collectors return null for min/max if no
+        // values are supplied
+        let arrow_schema = ArrowSchema::new(vec![
+            ArrowField::new("boolean", DataType::Boolean, false),
+            ArrowField::new("binary", DataType::Binary, false),
+            ArrowField::new("large_binary", DataType::LargeBinary, false),
+            ArrowField::new("string", DataType::Utf8, false),
+            ArrowField::new("large_string", DataType::LargeUtf8, false),
+        ]);
+        let schema = Schema::try_from(&arrow_schema).unwrap();
+        let mut collector = StatisticsCollector::new(&schema.fields);
+
+        // Collect stats
+        let id = schema.field("boolean").unwrap().id;
+        let builder = collector.get_builder(id).unwrap();
+        builder.append(StatisticsRow {
+            null_count: 0,
+            min_value: ScalarValue::Boolean(None),
+            max_value: ScalarValue::Boolean(None),
+        });
```

Comment:
@wjones127 should min/max in absence of boolean data be `false`/`true` or `None`?
https://lancedb.github.io/lance/format.html#statistic-values doesn't specifically say. This currently returns `None`.

Reply:
It should be `false/true`. We are storing the statistics in Lance format, and recall that we don't have null support for boolean data. So we cannot store None.